src/lib/auth/authconf.rb

```diff
@@ -0,0 +1,1089 @@
+            ldap_conf.chown(0, 0)
+            ldap_conf.write(ldap_make_conf)
+            ldap_conf.close
+            # If automount is enabled, overwrite openldap's ldap.conf as well.
+            if @ldap_nss.include?('automount')
+                ldap_conf = File.new('/etc/openldap/ldap.conf', 'w')
+                ldap_conf.chmod(644)
+                ldap_conf.chown(0, 0)
+                ldap_conf.write(ldap_make_conf)
+                ldap_conf.close
+            end
+            # Save PAM/NSS/daemon status
+            if @ldap_pam
+                Yast::Pam.Add('ldap')
+            else
+                Yast::Pam.Remove('ldap')
+            end
+            fix_pam
+            LDAP_CAPABLE_NSS_DBS.each { |db| nss_disable_module(db, 'ldap') }
+            if @ldap_nss.any?
+                @ldap_nss.each { |db| nss_enable_module(db, 'ldap') }
+            end
+        end
+
+        # Run ldapsearch to test the parameters. Return empty string if test is successful, otherwise return ldapsearch error output.
+        def ldap_test_bind(uri, start_tls, dn, password, base_dn)
+            # Make sure openldap client is installed
+            if !Yast::Package.Installed('openldap2-client')
+                if !Yast::Package.DoInstall(['openldap2-client'])
+                    return 'Failed to install openldap2-client package'
+                end
+            end
+            # Create a temporary file to hold the password
+            pwd_filename = "yastauthclient-ldaptestbind-#{Time.now.strftime('%Y%m%d%I%M%S')}"
+            pwd_file = File.open(pwd_filename, 'w', 0600)
+            pwd_file.write(password)
+            pwd_file.close
+            # Run ldapsearch with password bind
+            cmd = "ldapsearch -o nettimeout=5 -s one -x -H '#{uri}' "
+            if start_tls
+                cmd += '-ZZ '
+            end
+            if dn.to_s != ''
+                cmd += "-D '#{dn}' -y '#{pwd_filename}' "
+            end
+            cmd += "-b #{base_dn}"
+            out = ''
+            errout = ''
+            exitstatus = 0
+            Open3.popen3(cmd){ |stdin, stdout, stderr, control|
+                stdin.close
+                out = stdout.read
+                errout = stderr.read
+                exitstatus = control.value
+            }
+            File.unlink(pwd_file)
+            if exitstatus == 0
+                return ''
+            end
+            return "#{_('ERROR: ')} #{out}\n#{errout}"
+        end
+
+        # Parse and set Kerberos configuration
+        def krb_parse_set(content)
+            @krb_conf = KrbParse.parse(content)
+        end
+
+        # Load Kerberos PAM status, and configuration sections from /etc/krb5.conf.
+        def krb_read
+            begin
+                conf_file = File.new('/etc/krb5.conf')
+                content = conf_file.read
+            rescue Errno::ENOENT
+                log('Failed to read /etc/krb5.conf, the file is probably missing.')
```

Comment:
`log` does not work that way. See http://www.rubydoc.info/github/yast/yast-ruby-bindings/Yast/Logger

This can be made much simpler:
```rb
begin
  content = File.read('/etc/krb5.conf')
rescue Errno::ENOENT
  content = ''
end
krb_parse_set(content)
```